runtime/lua/vim/treesitter/dev.lua

```diff
@@ -76,7 +76,7 @@ end
 ---
 ---@package
 function TSTreeView:new(bufnr, lang)
-  local parser = vim.treesitter._get_parser(bufnr or 0, lang)
+  local parser = vim.treesitter.get_parser(bufnr or 0, lang, { error = false })
   if not parser then
     return nil, 'No parser available for the given buffer.'
   end
```

Comment:
can this be an assert (with context) as well?

Reply:
I think we should show an error in 
`:InspectTree`, but not here in the API.

Reply:
And I am actually not at all fond of propagating error messages as strings. Either
1. `assert` them,
2. notify them, or 
3. log them,

but keep them where they happen.

Reply:
> And I am actually not at all fond of propagating error messages as strings.

It's a standard Lua pattern. The error should be asserted/notified at the application level (`:InspectTree` in this case, or other plugin code), not the "library"/API level.

Reply:
Yes, and pcall is the standard approach for that pattern (throw/catch), isn't it?

Reply:
Sorry, I'm not making my point very well; let me try again. It's not about the error itself, it's about the message -- in particular, where it is constructed. I believe that this should be as close as possible to the point where the actual error is raised.

In this case, you're asking a function "do this thing", to which it replies "I cannot do this thing you ask". Now, we have just established a convention for that: simply return `nil`, which tells the calling function all it needs to know. Why should we return an additional string that says exactly the same thing, just in (more) words? What is that message telling the calling function that it cannot infer from the `nil` return, and what does it do with this information? Why can't `:InspectTree` not simply do `assert(parser, 'No parser available for the buffer')`?

(Or put differently: Hard-coded error strings without dynamic content are sus.)